ddl/backfilling.go

```diff
@@ -946,6 +1041,320 @@ func injectCheckBackfillWorkerNum(curWorkerSize int, isMergeWorker bool) error {
 	return nil
 }
 
+func addBatchBackfillJobs(sess *session, bfWorkerType backfillerType, reorgInfo *reorgInfo, notDistTask bool,
+	batchTasks []*reorgBackfillTask, bJobs []*BackfillJob, isUnique bool, id *int64) error {
+	bJobs = bJobs[:0]
+	instanceID := ""
+	if notDistTask {
+		instanceID = reorgInfo.d.uuid
+	}
+	// TODO: Adjust the number of ranges(region) for each task.
+	for _, task := range batchTasks {
+		bm := &model.BackfillMeta{
+			PhysicalTableID: reorgInfo.PhysicalTableID,
+			IsUnique:        isUnique,
+			EndInclude:      task.endInclude,
+			ReorgTp:         reorgInfo.Job.ReorgMeta.ReorgTp,
+			SQLMode:         reorgInfo.ReorgMeta.SQLMode,
+			Location:        reorgInfo.ReorgMeta.Location,
+			JobMeta: &model.JobMeta{
+				SchemaID: reorgInfo.Job.SchemaID,
+				TableID:  reorgInfo.Job.TableID,
+				Query:    reorgInfo.Job.Query,
+			},
+		}
+		bj := &BackfillJob{
+			ID:         *id,
+			JobID:      reorgInfo.Job.ID,
+			EleID:      reorgInfo.currElement.ID,
+			EleKey:     reorgInfo.currElement.TypeKey,
+			Tp:         bfWorkerType,
+			State:      model.JobStateNone,
+			InstanceID: instanceID,
+			CurrKey:    task.startKey,
+			StartKey:   task.startKey,
+			EndKey:     task.endKey,
+			Meta:       bm,
+		}
+		*id++
+		bJobs = append(bJobs, bj)
+	}
+	if err := AddBackfillJobs(sess, bJobs); err != nil {
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+func (*ddlCtx) splitTableToBackfillJobs(sess *session, reorgInfo *reorgInfo, pTbl table.PhysicalTable, isUnique bool,
+	bfWorkerType backfillerType, startKey kv.Key, currBackfillJobID int64) error {
+	endKey := reorgInfo.EndKey
+	isFirstOps := true
+	bJobs := make([]*BackfillJob, 0, genTaskBatch)
+	for {
+		kvRanges, err := splitTableRanges(pTbl, reorgInfo.d.store, startKey, endKey)
```

Comment:
Is it necessary to split the same range again and again? Can we move this out of the loop?

Reply:
For this `genTaskBatch` batch processing, wait until "job count < `minGenTaskBatch`" will batch acquisition tasks, here may take a long time, then the regions may be reallocated